### PR TITLE
[Update] Add eval function of tempoGAN example and update doc

### PR DESCRIPTION
**Pull request by lijialin03**
### PR types

Others
### PR changes

Others
### Describe

1. Add eval function of tempoGAN example
2. Update doc of tempoGAN example

### CODE REVIEW

File: docs/zh/examples/tempoGAN.md [hunk 1]
```diff
@@ -291,16 +291,57 @@ examples/tempoGAN/tempoGAN.py:237:248
...
+examples/tempoGAN/tempoGAN.py:297:313
+--8<--
+```
+
 具体代码请参考 [完整代码](#4) 中 tempoGAN.py 文件。
 
+#### 3.10.2 评估器构建
+
+在训练过程中通常会按一定轮数间隔，用验证集（测试集）评估当前模型的训练情况，进行评估时，对所有验证集数据都进行，因此将 `eval_dataloader_cfg`（验证集dataloader配置，构造方式与 `train_dataloader_cfg` 类似）传递给`ppsci.validate.SupervisedValidator` 来构造评估器：
+
```
Comment (HydrogenSulfate) [inline docs/zh/examples/tempoGAN.md:314]: 删除“进行评估时，对所有验证集数据都进行”这句话

File: docs/zh/examples/tempoGAN.md [hunk 2]
```diff
@@ -23,7 +23,7 @@
     # windows
     # curl https://paddle-org.bj.bcebos.com/paddlescience/datasets/tempoGAN/2d_train.mat --output datasets/tempoGAN/2d_train.mat
     # curl https://paddle-org.bj.bcebos.com/paddlescience/datasets/tempoGAN/2d_valid.mat --output datasets/tempoGAN/2d_valid.mat
-    python tempoGAN.py mode=eval EVAL.pretrained_model_path=https://paddle-org.bj.bcebos.com/paddlescience/models/tempogan/tempogan_pretrained.pdparams
+    python tempoGAN.py mode=eval EVAL.pretrained_model_path=https://paddle-org.bj.bcebos.com/paddlescience/models/tempoGAN/tempogan_pretrained.pdparams
```
Comment (HydrogenSulfate) [inline docs/zh/examples/tempoGAN.md:26]: 模型训练和推理命令中的数据集wget参数不正确，应该用：

`wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/tempoGAN/2d_valid.mat -P datasets/tempoGAN/`

`wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/tempoGAN/2d_train.mat -P datasets/tempoGAN/`

否则没有文件夹会直接报错

File: examples/tempoGAN/conf/tempogan.yaml [hunk 3]
```diff
@@ -90,3 +91,7 @@ TRAIN:
 # evaluation settings
 EVAL:
   pretrained_model_path: null
+  eval_with_no_grad: true
+  batch_size:
+    sup_validator: 8
+  save_outs: true
```
Comment (HydrogenSulfate) [inline examples/tempoGAN/conf/tempogan.yaml:97]: 末尾换行

File: examples/tempoGAN/tempoGAN.py [hunk 4]
```diff
@@ -315,6 +313,103 @@ def evaluate(cfg: DictConfig):
...
+                img_target, img_pred
+            )
+            eval_mse_list.append(eval_mse)
+            eval_psnr_list.append(eval_psnr)
+            eval_ssim_list.append(eval_ssim)
+
+        if cfg.EVAL.save_outs:
+            from matplotlib import image as Img
+
+            os.mkdir(f"{cfg.output_dir}eval_outs/")
```
Comment (HydrogenSulfate) [inline examples/tempoGAN/tempoGAN.py:360]: 1. 用`os.makedirs(PATH, exist_ok=True)`代替os.mkdir
2. 路径拼接使用osp.join

File: examples/tempoGAN/tempoGAN.py [hunk 5]
```diff
@@ -315,6 +313,103 @@ def evaluate(cfg: DictConfig):
...
+            for i, image in enumerate(output_arr):
+                Img.imsave(
+                    f"{cfg.output_dir}eval_outs/out_{i}.png",
```
Comment (HydrogenSulfate) [inline examples/tempoGAN/tempoGAN.py:363]: 同上，`osp.join`

File: examples/tempoGAN/tempoGAN.py [hunk 6]
```diff
@@ -315,6 +313,103 @@ def evaluate(cfg: DictConfig):
...
+                    image,
+                    vmin=0.0,
+                    vmax=1.0,
+                    cmap="gray",
+                )
+        logger.message(
+            f"MSE: {np.mean(eval_mse_list)}, PSNR: {np.mean(eval_psnr_list)}, SSIM: {np.mean(eval_ssim_list)}"
+        )
+        return {"dummy": 0.0}
```
Comment (HydrogenSulfate) [inline examples/tempoGAN/tempoGAN.py:372]: 训练结束的指标评估和可视化可以不一定用内置的Validator或Visualizer，可以写在 `solver.train()` 结束后的下方，加上合适的注释说明代码即可

File: examples/tempoGAN/tempoGAN.py [hunk 7]
```diff
@@ -315,6 +313,103 @@ def evaluate(cfg: DictConfig):
...
+            "shuffle": False,
+        },
+    }
+    sup_validator = ppsci.validate.SupervisedValidator(
+        {**eval_dataloader_cfg, "batch_size": cfg.EVAL.batch_size.sup_validator},
+        ppsci.loss.MSELoss("mean"),
+        {"density_high": lambda out: out["output_gen"]},
+        metric={"metric": ppsci.metric.FunctionalMetric(metric_func_eval)},
+        name="sup_validator_gen",
+    )
```
Comment (HydrogenSulfate) [inline examples/tempoGAN/tempoGAN.py:395]: 如上所述，可以拆出来，不一定要做成sup_validator

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: docs/zh/examples/tempoGAN.md
Comment (HydrogenSulfate) [inline docs/zh/examples/tempoGAN.md:1]: 1. 删除末尾逗号，这样字典会被格式化成一行
image
2. 3.7 可视化器构建 第二段代码定位不正确
image
3. 文字描述与下方代码无关
image
4. 开头添加预训练模型和对应的评估指标
image

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

**Review by HydrogenSulfate — APPROVED** [review]
LGTM
